### DISCUSSION TITLE
Logging?

### DISCUSSION THREAD

**Discussion by swlkr** [swlkr/osprey#11] [Ideas]
Should osprey include web request/response logging or not?

**Comment by pepe** [comment]
I would say yes. I see Osprey all-in-one tool for creating the web application, and web applications need to log from time to time.

Internal logging has some benefits:

- not only "automatic" logs, but you can also be able to coopt the logging when you are debugging, passing critical place or panicking
- know the state of the place it is logging from, in before hooks or in the dispatched function
- and yes know the internals without explicitly stated so that you can have a request, response or body at hand

So theoretical API goes like this:

- macro for enabling automatic standard logging `(enable :logging)`
- way to influence what and how something will be logged `(dyn :format)`?
- general logging function `(log "Oh yes!")`
- state aware debug function `(log-state :body :params)`

As always, I could help. At least as a tester.

**Comment by swlkr** [comment]
That api is interesting, you could collapse it a bit more, like:

```clojure
(enable :logging)

(log "Oh yes!")
(log :body :params) # => "body: {} params: {}"
```

Not sure how the log format could be changed though

**Reply by pepe** [reply]
Oh, yes, this is better. I forgot we are using the best lang, so there are some affordances. Your log is definitely much better.

Please scratch the idea of the format. It is not well thought out. And frankly, not needed. 

Thank you!
